src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -3066,6 +3066,13 @@ VSTGUI::COptionMenu *SurgeGUIEditor::makeTuningMenu(VSTGUI::CRect &menuRect)
                                 }
                             }
                             auto sc = Surge::Storage::readSCLFile(sf);
+
+                            if (sc.count <= 0) // invalid
```

Comment:
This is a smart check!

But  SurgeStorage::retuneToScale is called several places. I would rather put this check inside the retuneToScale and add a Scale.isValid() method which for now just checks count > 0 but later could check more.

Do you mind doing that change? I'm happy to add it to my list if you aren't. But these are all pretty clean classes in src/common/Tuning

Reply:
Oh and retuneToScale should just start with if( ! scale.isValid() ) { UserInteractions... return } and leave the state unchanged. Thanks!

Reply:
Ok I'll do this

Reply:
Thank you very much!